scripts/mutect2_wdl/mutect2.wdl

```diff
@@ -22,32 +22,57 @@ task M2 {
   File tumor_bam
   File tumor_bam_index
   String tumor_sample_name
-  File normal_bam
-  File normal_bam_index
-  String normal_sample_name
+  File? normal_bam
+  File? normal_bam_index
+  String? normal_sample_name
   File? pon
   File? pon_index
   File? dbsnp
   File? dbsnp_index
   File? cosmic
   File? cosmic_index
+  String output_vcf_name
 
   command {
+  if [[ "$normal_bam" == *.bam ]]; then
+      normal_command_line="-I ${normal_bam} -normal ${normal_sample_name}"
+  fi
+
     java -Xmx4g -jar ${gatk4_jar} Mutect2 \
     -R ${ref_fasta} \
     -I ${tumor_bam} \
-    -I ${normal_bam} \
     -tumor ${tumor_sample_name} \
-    -normal ${normal_sample_name} \
+    $normal_command_line \
     ${"--dbsnp " + dbsnp} \
     ${"--cosmic " + cosmic} \
     ${"--normal_panel " + pon} \
     -L ${intervals} \
-    -O ${tumor_sample_name}-vs-${normal_sample_name}.vcf
+    -O "${output_vcf_name}.vcf"
+  }
+
+  output {
+    File output_vcf = "${output_vcf_name}.vcf"
+  }
+}
+
+# HACK: cromwell can't handle the optional normal sample name in output or input --
+# string interpolation of optionals only works inside a command block
+# thus we use this hack
+task MakeOutputVcfName {
```

Comment:
I could do the alternative, but the way it works now allows you to either leave the normal fields blank or put in some placeholder like `TUMOR_ONLY` or `TEMPORARY_FORMATTING_HACK`.  I'm loathe to make an empty field the only option because the difference between the correct thee-tabs-and-nothing-else format and an incorrect format could be invisible.

It's true that it's awkward to pass both the bam and the sample name to this task, but since it's not exposed to the user I'm comfortable with the redundancy.

I will make `normal_bam` a `String`.  Do you mean just for this particular task?